test/e2e/autotls/auto_tls_test.go

```diff
@@ -91,6 +93,41 @@ func testAutoTLS(t *testing.T) {
 	rootCAs := createRootCAs(t, clients, objects.Route.Namespace, certName)
 	httpsClient := createHTTPSClient(t, clients, objects, rootCAs)
 	testingress.RuntimeRequest(t, httpsClient, "https://"+objects.Service.Status.URL.Host)
+
+	t.Run("Tag route", func(t *testing.T) {
+		// Verify that a certificate is created when route is tagged
+		if _, err := v1test.UpdateServiceRouteSpec(t, clients, names, servingv1.RouteSpec{
+			Traffic: []servingv1.TrafficTarget{{
+				Tag:            "tag1",
+				Percent:        ptr.Int64(100),
+				LatestRevision: ptr.Bool(true),
+			}},
+		}); err != nil {
+			t.Fatalf("Failed to update Service route spec: %v", err)
+		}
+		// Wait for ingress TLS configuration.
+		if err = v1test.WaitForServiceState(clients.ServingClient, names.Service, v1test.IsServiceReady, "ServiceIsReady"); err != nil {
+			t.Fatalf("Service %s did not become ready: %v", names.Service, err)
+		}
+
+		route, err := clients.ServingClient.Routes.Get(objects.Route.Name, metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("Failed to retrieve route: %v", err)
+		}
+
+		certificates, err := clients.NetworkingClient.Certificates.List(metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("Failed to list certificates: %v", err)
+		}
+		expectedDomains := []string{route.Status.URL.Host, "tag1-" + route.Status.URL.Host}
+		var certDomains []string
+		for _, certificate := range certificates.Items {
+			certDomains = append(certDomains, certificate.Spec.DNSNames...)
+		}
+		if !cmp.Equal(expectedDomains, certDomains) {
```

Comment:
We probably want to check if the `tag` URL is accessible by sending a HTTPS request to the tag URL.

Reply:
I'll add that.